app/src/main/java/fr/neamar/kiss/loader/LoadContactsPojos.java

```diff
@@ -53,8 +54,24 @@ public LoadContactsPojos(Context context) {
                             .getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY));
                     contact.timesContacted = Integer.parseInt(cur.getString(cur
                             .getColumnIndex(ContactsContract.CommonDataKinds.Phone.TIMES_CONTACTED)));
-                    contact.setName(cur.getString(cur
-                            .getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+                    contact.setName(cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+
+                    String contactID = cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.CONTACT_ID));
+
+                    Cursor cur2 = context.getContentResolver().query(
```

Comment:
Unfortunately not, this was actually the first thing I tried. For some reason usernames are stored in a different table. If I try to get it from the main query, I just get a contact's phone number.  
Maybe it could be possible to do a second query right after the main one and simply iterate it instead of having a query for every contact (contacts without nickname would return it null or empty string)? Or even a simple sql JOIN should maximize performance - but I am not sure how to do it.